config/defaults.go

```diff
@@ -32,9 +32,10 @@ var defaultConfig = agentconfig.AgentConfig{
 			wrapperspb.String("x-www-form-urlencoded")},
 	},
 	Reporting: &agentconfig.Reporting{
-		Endpoint:          agentconfig.String("http://localhost:9411/api/v2/spans"),
-		Secure:            agentconfig.Bool(false),
-		TraceReporterType: agentconfig.TraceReporterType_ZIPKIN,
-		CertFile:          agentconfig.String(""),
+		Endpoint:                agentconfig.String("http://localhost:9411/api/v2/spans"),
+		Secure:                  agentconfig.Bool(false),
+		TraceReporterType:       agentconfig.TraceReporterType_ZIPKIN,
+		CertFile:                agentconfig.String(""),
+		EnableGrpcLoadbalancing: agentconfig.Bool(false),
```

Comment:
My reasoning for false by default is along the same reasoning we opted to have a config at all, unexpected deployment, or the dns resolver unexpectedly not working in some environments.

I do think it's nice since this allows us to mimic the defaults used by grpc-go(where passthrough is the default resolver used instead of dns, which is what this flag enables)

Additionally, if we use true as default, users won't actually get any value unless they configure the service to run as headless as well so I lean towards both configurations should be done as opt-in intentional changes to defaults.

I'm definitely open to having this default to true,  I'm mostly concerned with adding any initial install friction if for some reason the dns resolver doesn't work OOTB for a user env

Thoughts? 

cc @ryanericson